sdk/core/azure-core-http-vertx/src/test/java/com/azure/core/http/vertx/VertxHttpClientLocalTestServer.java

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.http.vertx;
+
+import com.azure.core.test.http.LocalTestServer;
+import org.eclipse.jetty.util.Callback;
+
+import javax.servlet.ServletException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * {@link LocalTestServer} used by all tests in this package.
+ */
+public final class VertxHttpClientLocalTestServer {
+    private static volatile LocalTestServer server;
+    private static final Semaphore SERVER_SEMAPHORE = new Semaphore(1);
+
+    private static volatile LocalTestServer proxyServer;
+    private static final Semaphore PROXY_SERVER_SEMAPHORE = new Semaphore(1);
+
+    public static final String GET_ENDPOINT = "/get";
+    public static final byte[] EXPECTED_GET_BYTES = new byte[32768];
+
+    public static final byte[] SHORT_BODY = "hi there".getBytes(StandardCharsets.UTF_8);
+    public static final byte[] LONG_BODY = createLongBody();
+    public static final String RETURN_HEADERS_AS_IS_PATH = "/returnHeadersAsIs";
+
+    public static final String PROXY_USERNAME = "foo";
+    public static final String PROXY_PASSWORD = "bar";
+
+    public static final String SERVICE_ENDPOINT = "/default";
+
+    static {
+        ThreadLocalRandom.current().nextBytes(EXPECTED_GET_BYTES);
+    }
+
+    /**
+     * Gets the {@link LocalTestServer} instance.
+     *
+     * @return The {@link LocalTestServer} instance.
+     */
+    public static LocalTestServer getServer() {
```

Comment:
it looks like all <Client>LocalTestServer have a lot in common, would it make sense to move common methods to LocalTestServer  or somewhere else in the azure-core-test?